packages/solid-js/src/createFireproof.tsx

```diff
@@ -0,0 +1,142 @@
+ * @returns An object containing the custom database accessor and database scoped `createLiveQuery` and `createDocument` hooks
+ *
+ *
+ * ### Overview
+ *
+ * TL;DR: Only use this hook if you need to configure a database name other than the default `@fireproof/db`.
+ *
+ * For most applications, using the `createLiveQuery` or `createDocument` hooks exported from `@fireproof/solid-js` should
+ * suffice for the majority of use-cases. Under the hood, they act against a database named `@fireproof/db` instantiated with
+ * default configurations. However, if you need to do a custom database setup or configure a database name more to your liking
+ * than the default `@fireproof/db`, then use `createFireproof` as it exists for that purpose. It will provide you with the
+ * custom database accessor and *lexically scoped* versions of `createLiveQuery` and `createDocument` that act against said
+ * custom database.
+ *
+ * If you need to, using the power of Solid, you can distribute the custom database accessor and lexical hooks through your
+ * app using the Context APIs or just simply instantiating them globally and importing it where you need it like you would with any global
+ * SolidJS signal.
+ */
+export function createFireproof(
+  name: string | Database = "@fireproof/db",
+  config: FireproofOptions = {}
+): CreateFireproof {
+  // The database connection is cached, so subsequent calls to fireproof with the same name will
+  // return the same database object. This makes it safe to call fireproof with every render.
+  const database = createMemo(() => (typeof name === "string" ? fireproof(name, config) : name));
+
+  function createDocument(initialDoc: Doc): CreateDocumentResult {
+    const [doc, setDoc] = createSignal(initialDoc);
+
+    const updateDoc = (newDoc?: Doc, options: UpdateDocFnOptions = { replace: false, deepMerge: false }) => {
+      if (!newDoc) return setDoc(initialDoc);
+      if (options.replace) return setDoc(newDoc);
+
+      return setDoc((d) => (options.deepMerge ? deepmerge(d, newDoc) : { ...d, ...newDoc }));
+    };
+
+    const saveDoc = async () => {
+      const putDoc = initialDoc._id ? ({ ...doc(), _id: initialDoc._id } as Doc) : doc();
+      return await database().put(putDoc);
+    };
+
+    const refreshDoc = async (db: Database) => {
+      if (!initialDoc._id) return;
+
+      // TODO: add option for mvcc checks
+      setDoc(await db.get(initialDoc._id).catch(() => initialDoc));
+    };
+
+    createEffect(() => {
+      const db = database();
+      const unsubscribe = db.subscribe((changes) => {
+        if (changes.find((c) => c.key === initialDoc._id)) {
+          refreshDoc(db); // TODO: use change.value
+        }
+      });
+
+      onCleanup(() => {
+        unsubscribe();
+      });
+    });
+
+    // TODO: Is this effect necessary? It seems like it would be redundant with the above effect perhaps?
+    createEffect(() => {
+      void refreshDoc(database());
+    });
```

Comment:
This portion sticks out to me because I'm questioning whether we need the 2nd `createEffect` that calls `refreshDoc`. Is the one being called from within the first `createEffect` sufficient and can we do away with the second one?